src/c_lib/lib/angular_momentum/wigner_element.c

```diff
@@ -16,9 +16,12 @@ double SQRT_3 = 1.732050807568877;
 static inline double fac(double x) {
   if (x < 0) exit(1);
 
+  if (x == 0.0) return 1;
+
   int ix = (int)x;
   double sum = 1.0;
-  while (ix-- > 1) sum *= ix;
+  do sum *= ix;
+  while (--ix > 1);
```

Comment:
Replacing 
```py
while (ix-- > 1) sum *= ix;
``` 
with 

```py
while (ix > 1) sum *= ix--;
```

should fix the (x-1)! and x! bug. There is no need for explicit (x==0) condition.